### PR TITLE
[HLS-2615] Add helios instrumentation indication to logback and log4j

### PR DESCRIPTION
**Pull request by YochayCO**

### CODE REVIEW

File: instrumentation/log4j/log4j-mdc-1.2/javaagent/src/test/groovy/Log4j1MdcTest.groovy [hunk 1]
```diff
@@ -33,6 +33,8 @@ class Log4j1MdcTest extends AgentInstrumentationSpecification {
     events[1].getMDC("trace_id") == null
     events[1].getMDC("span_id") == null
     events[1].getMDC("trace_flags") == null
+
+    assertTraces(0) {}
```
Comment (LironKS) [inline instrumentation/log4j/log4j-mdc-1.2/javaagent/src/test/groovy/Log4j1MdcTest.groovy]: what's this assertion?

File: instrumentation/logback/logback-mdc-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/logback/mdc/v1_0/LoggingEventInstrumentation.java [hunk 2]
```diff
@@ -55,6 +57,14 @@ public void transform(TypeTransformer transformer) {
   @SuppressWarnings("unused")
   public static class GetMdcAdvice {
 
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void onEnter() {
+      Context spanContext = Java8BytecodeBridge.currentContext();
+      Span span = Java8BytecodeBridge.spanFromContext(spanContext);
+      span.setAttribute(HELIOS_INSTRUMENTED_INDICATION, "logback");
+      System.out.print(span);
```
Comment (LironKS) [inline instrumentation/logback/logback-mdc-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/logback/mdc/v1_0/LoggingEventInstrumentation.java]: remove this

File: instrumentation/logback/logback-mdc-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/logback/mdc/v1_0/LoggingEventInstrumentation.java [hunk 3]
```diff
@@ -69,11 +79,16 @@ public static void onExit(
...
       }
 
-      SpanContext spanContext = Java8BytecodeBridge.spanFromContext(context).getSpanContext();
+      Span span = Java8BytecodeBridge.spanFromContext(context);
+      SpanContext spanContext = span.getSpanContext();
       if (!spanContext.isValid()) {
         return;
       }
 
+      if (span.isRecording()) {
```
Comment (LironKS) [inline instrumentation/logback/logback-mdc-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/logback/mdc/v1_0/LoggingEventInstrumentation.java:80]: why we are setting the attribute both on method enter and on method exist?

File: instrumentation/log4j/log4j-mdc-1.2/javaagent/src/test/groovy/Log4j1MdcTest.groovy (hunk 1 above)
Comment (YochayCO) [reply instrumentation/log4j/log4j-mdc-1.2/javaagent/src/test/groovy/Log4j1MdcTest.groovy]: I guess it's not necessary, just made sure there are zero traces in this test

File: instrumentation/logback/logback-mdc-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/logback/mdc/v1_0/LoggingEventInstrumentation.java (hunk 3 above)
Comment (YochayCO) [reply instrumentation/logback/logback-mdc-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/logback/mdc/v1_0/LoggingEventInstrumentation.java:80]: My bad, removing the one on the onEnter

**Review by LironKS — APPROVED** [review]

**Pull request merged**